jetty-annotations/src/main/java/org/eclipse/jetty/annotations/ContainerInitializerAnnotationHandler.java

```diff
@@ -72,13 +84,14 @@ public void handle(FieldInfo info, String annotationName)
     @Override
     public void handle(MethodInfo info, String annotationName)
     {
-        if (annotationName == null || !_annotation.getName().equals(annotationName))
+        if (!_annotation.getName().equals(annotationName))
             return;
-        _initializer.addAnnotatedTypeName(info.getClassInfo().getClassName());
+        _holder.addStartupClasses(info.getClassInfo().getClassName());
     }
 
+    @Deprecated
     public ContainerInitializer getContainerInitializer()
     {
-        return _initializer;
+        return null;
```

Comment:
If we want to still allow the constructor giving a `ContainerInitializer` we might not want to break this.

Maybe we could store the `ContainerInitializer` as a private field so it will only be set if you use the `ContainerInitializerAnnotationHandler(ContainerInitializer initializer, Class<?> annotation)` constructor.

Reply:
See my comment above.